sky/spot/spot_utils.py

```diff
@@ -231,14 +231,17 @@ def stream_logs_by_id(job_id: int, follow: bool = True) -> str:
     backend = backends.CloudVmRayBackend()
     spot_status = spot_state.get_status(job_id)
     while not spot_status.is_terminal():
-        if spot_status != spot_state.SpotStatus.RUNNING:
+        handle = global_user_state.get_handle_from_cluster_name(cluster_name)
+        # Check the handle: The cluster can be removed from the table before the
+        # spot state is updated by the controller. In this case, we should skip
+        # the logging, and wait for the next round of status check.
+        if handle is None or spot_status != spot_state.SpotStatus.RUNNING:
```

Comment:
It's possible that `handle is None and spot_status == RUNNING`. In this case the message below may be a bit weird. How about simply print "INFO: The log is not ready yet. Waiting for ..." in that case, or something better?

Reply:
Good catch! Added the handling for those cases. Thanks!